### PR TITLE
Use transactions for tree manipulations

### PR DESCRIPTION
**Pull request by svenseeberg**
Apply changes to MPTT tree structures in SQL tables with transactions to prevent race conditions with other requests.

### Short description
Use the `@transaction.atomic` decorator for all views that update MPTT trees.

Fixes: #1320 (maybe)

### CODE REVIEW

File: integreat_cms/cms/views/pages/page_form_view.py [hunk 1]
```diff
@@ -176,6 +177,7 @@ def get(self, request, *args, **kwargs):
             },
         )
 
+    @transaction.atomic
```
Comment (timoludwig) [inline integreat_cms/cms/views/pages/page_form_view.py:180]: Maybe let's add this to the post-function of `LanguageTreeNodeFormView` as well?

**Review by timoludwig — APPROVED** [review]
Thanks! :+1:
Let's try it out!

File: integreat_cms/cms/views/pages/page_form_view.py (hunk 1 above)
Comment (svenseeberg) [reply integreat_cms/cms/views/pages/page_form_view.py:180]: Yes, I missed that one.

**Review by timoludwig — APPROVED** [review]
:tada:

**Pull request merged**